recipes/_start.rb

```diff
@@ -10,6 +10,16 @@
   not_if { platform_family?('windows') }
 end
 
+ruby_block 'insert ulimit' do
+  block do
+    file = Chef::Util::FileEdit.new('/etc/init.d/splunk')
+    file.insert_line_after_match(/^RETVAL=\d$/, "ulimit -n #{node['splunk']['limits']['open_files']}")
+    file.write_file
+  end
+  not_if { platform_family?('windows') || File.readlines('/etc/init.d/splunk').grep(/ulimit/).any? }
```

Comment:
What if we want to adjust the value of the ulimit?  If the current value is 8196 and I update my config to a different value, this logic won't every modify the ulimit in the init.d file again.